tsn/src/bin/latency.rs

```diff
@@ -0,0 +1,430 @@
+                    let elapsed_ns = tdiff.tv_sec * 1000000000 + tdiff.tv_nsec;
+                    println!(
+                        "RTT: {}.{:03} µs ({} → {})",
+                        elapsed_ns / 1000,
+                        elapsed_ns % 1000,
+                        tstart.tv_nsec,
+                        tend.tv_nsec
+                    );
+                } else {
+                    println!("TIMEOUT: -1µs ({} -> N/A)", tstart.tv_nsec);
+                }
+            }
+            if !precise {
+                request.tv_sec = 0;
+                request.tv_nsec =
+                    700 * 1000 * 1000 + (rand::thread_rng().gen_range(0..32767) as i64 % 10000000);
+                libc::nanosleep(&request, std::ptr::null_mut::<libc::timespec>());
+            }
+        }
+    }
+}
+
+fn main() {
+    unsafe {
+        let arguments = std::env::args();
+        let arguments = arguments::parse(arguments).unwrap();
+        let mut args: Arguments = Arguments {
+            ..Default::default()
+        };
+
+        let mode = arguments
+            .get::<String>("mode")
+            .expect("Need mode to run server or client");
+        if mode == "s" {
+            args = Arguments {
+                verbose: arguments
+                    .get::<bool>("verbose")
+                    .expect("Need verbose on/off"),
+                iface: arguments
+                    .get::<String>("interface")
+                    .expect("Need interface to use"),
+                mode,
+                target: String::from(""),
+                count: 0,
+                size: arguments.get::<i32>("size").expect("Need packet size"),
+                precise: false,
+                oneway: arguments
+                    .get::<bool>("oneway")
+                    .expect("Check latency on receiver side"),
+            };
+        } else if mode == "c" {
+            args = Arguments {
+                verbose: arguments
+                    .get::<bool>("verbose")
+                    .expect("Need verbose on/off"),
+                iface: arguments
+                    .get::<String>("interface")
+                    .expect("Need interface to use"),
+                mode,
+                target: arguments
+                    .get::<String>("target")
+                    .expect("Need target MAC address"),
+                count: arguments
+                    .get::<i32>("count")
+                    .expect("Need count to send packet"),
+                size: arguments.get::<i32>("size").expect("Need packet size"),
+                precise: arguments
+                    .get::<bool>("precise")
+                    .expect("Send packet at precise 0ns or not"),
+                oneway: arguments
+                    .get::<bool>("oneway")
+                    .expect("Check latency on receiver side"),
+            };
+        } else {
+            println!("Unknown mode");
+        }
+
+        SOCK = tsn::tsn_sock_open(
```

Comment:
ALLCAPS naming is for constant value

Reply:
Rust compiler warns that global variables don't have ALLCAPS names